### ISSUE TITLE
Modal post doesn't open inside index route but it does in different routes

### ISSUE THREAD

**Issue by roelmagdaleno** [nuxt/nuxt.js#5593]
### Version

[v2.6.2](https://github.com/nuxt.js/releases/tag/v2.6.2)

### Reproduction link

[https://codesandbox.io/s/2v4vyr850n](https://codesandbox.io/s/2v4vyr850n)

### Steps to reproduce

You have to go to these routes (or click the links in the index route):

`/timewall` - works
`/testing` - works
`/` - not working

After that, you need to click on the post card and it will open the modal. This won't happen inside the index route and will show you the error.

### What is expected ?

Open the post modal inside `/timewall` and `/`

### What is actually happening?

Trying to open the modal in index route, will throw this error:

`_vm.$refs.postModal is undefined`

### Additional comments?

I configured my nuxt.config.js file to add to the previous routes a children so it can access the `post/_slug.vue` template inside the routes. 

![2019-04-24 00_47_10-web-app [F__Desarrollos_Doki_web-app] - ..._nuxt.config.js [web-app] - PhpStorm.png](https://cmty.s3.eu-west-1.amazonaws.com/71995937/966fdbf0-6654-11e9-bc99-8390039d252a.png)

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9096">#c9096</a>)</em></sub></div>

**Comment by clarkdo** [comment]
@roelmagdaleno 

In you case, `/post/_slug` is child of `/testing`, but `/post/_slug` is a root level path, it's not a suggested structure.

I suggest that you can remove `/` of `path: "/post/:slug"` in `extendRoutes` of `nuxt.config.js` and leverage lifecycle method to control the modal.

```js
// nuxt.config.js
path: "post/:slug",

// timewall.vue

// remove click event
-- @click.prevent="$refs.postModal.show()"

// control modal in mounted
  mounted() {
    if (this.$route.name.startWith('modal-post-from-')) {
      this.$refs.postModal.show();
    }
  },
```

**Comment by roelmagdaleno** [comment]
Hey @clarkdo, thanks for your solution.

I tested it in *v2.6.0* and works perfectly, but in more updated versions (*v2.8.0*) doesn't work, it doesn't open the modal.

So what I did to fix it was to add a watcher inside my `Timewall.vue` component:

````
watch: {
        '$route': function(to) {
            if (to.name.startsWith("modal-post-from-")) {
                this.$refs.postModal.show();
            }
        }
    },
````

The code is similar as you posted it, and that fixed it. But now I got another issue, when I refreshed the page that contains the modal, it didn't go to its own component, it was opening the index or timewall routes because the `/post/:slug` route was after the `/timewall` and `index` routes.

So I had to use `@nuxtjs/router` module to create custom routes and move `/post/:slug` before `/timewall` and `index`.

I don't know if that's the correct way to fix it, but after that it works like a charm!

Thanks :)
